add_sourceapp.sh

```diff
@@ -139,6 +139,14 @@ $notinzip";;
   fi
   echo "APK is complete, certificate is valid and signed by Google"
 
+  checkpermissions "$apk"
+  checked="$?"
```

Comment:
Checking the return-code here is not very easy to read/understand.

I think it would be better to handle both the checking and any errors in the checkpermission() function itself.